api/v1alpha1/storageconsumer_types.go

```diff
@@ -42,7 +42,18 @@ type StorageConsumerSpec struct {
 	Enable bool `json:"enable,omitempty"`
 	// StorageQuotaInGiB describes quota for the consumer
 	// +optional
-	StorageQuotaInGiB int `json:"storageQuotaInGiB,omitempty"`
+	StorageQuotaInGiB          int         `json:"storageQuotaInGiB,omitempty"`
+	RadosNamespace             string      `json:"radosNamespace,omitempty"`
+	SubVolumeGroup             string      `json:"subVolumeGroup,omitempty"`
+	StorageClasses             []ClassSpec `json:"storageClasses,omitempty"`
+	VolumeSnapshotClasses      []ClassSpec `json:"volumeSnapshotClasses,omitempty"`
+	VolumeGroupSnapshotClasses []ClassSpec `json:"volumeGroupSnapshotClasses,omitempty"`
+	ClientProfiles             []string    `json:"clientProfiles,omitempty"`
```

Comment:
@leelavg I believe we agreed that ClientProfiles are not admin configurable. Why do we have this as part of the consumer spec?

Reply:
I'm not sure, we decided to remove the secret (provisioner, node) aliases from spec but not clientProfiles.

Nevertheless, if we already know the storageconsumer upgrade mode we can figure these out, will update.

Reply:
In one of the meetings, we were talking about having the clientProfile name something similar to storageID (ceph_FSID,radosnamespace) and (ceph_FSID,subvolumegroupName). If we are going with that shouldn't we have two different clientProfiles? one for rbd and another for cephfs?

Reply:
This was in context when @Madhu-1 raised a point wrt MDR but it was decided that StorageConsumerUID will always be unique (for all intents & purposes) and could be used as clusterID and subsequently as clientProfile, isn't it?

Reply:
This is all depends on the clientProfile naming scheme. Do we have a PR in place where we are calculating the clientProfile names?

Reply:
no, didn't we settle w/ storageconsumer uid as they are unique? I don't remember we are calculating anything.

Reply:
@leelavg @Madhu-1 @rewantsoni 
Going forward we have a single scheme as discussed and agreed and captured on the weekly call meeting notes.
We will support the older schemes based on the indicator that marks an updated internal or updated provider.
In all cases, this is not an admin nob